### ISSUE TITLE
Fix URL Escaping

### ISSUE THREAD

**Issue by chrisjsewell** [ExecutableBookProject/markdown-it-py#1]
This is the only point of non-compliance with markdown-it.
See `tests/test_cmark_spec/test_spec.py` and `tests/test_port/test_fixtures.py` for details

**Comment by chrisjsewell** [comment]
An example of this issue is going from this:

```
Checking combination of replaceEntities and unescapeMd:', input = '~~~ &amp;&bad;\\&amp;\\\\&amp;\njust a funny little fence\n~~~\n
```

To this:

```html
<pre><code class="&amp;&amp;bad;&amp;amp;\\&amp;">just a funny little fence\n</code></pre>\n
```

which at present is going to this:

```html
<pre><code class="language-&amp;amp;&amp;bad;\&amp;amp;\\&amp;amp;">just a funny little fence\n</code></pre>\n
```

**Comment by chrisjsewell** [comment]
The portability issue lies in the fact that markdown-it uses the external JS dependancies [mdurl](https://www.npmjs.com/package/mdurl) and [punycode](https://www.npmjs.com/package/punycode) for their functions, which are obviously not directly useable for python. Currently we use use the built-in pythoon `html.escape` and `quote` functions for this
https://github.com/ExecutableBookProject/markdown-it-py/blob/3140d46ac359ae76324b422970c558e9be85a83b/markdown_it/normalize_url.py#L58
https://github.com/ExecutableBookProject/markdown-it-py/blob/3140d46ac359ae76324b422970c558e9be85a83b/markdown_it/common/utils.py#L191

but clearly there is some niggling issues to finesse.

Could also loo into the mistletoe, commonmark-py , etc code for inspiration

**Comment by chrisjsewell** [comment]
Also within this issue we shoulf make sure of HTML security, as discussed here; https://github.com/markdown-it/markdown-it/blob/master/docs/security.md#security

**Comment by chrisjsewell** [comment]
Partially fixed in 124297c6ed36cb7f49078d4a42be4db0ddb74c5b

**Comment by chrisjsewell** [comment]
closed in 22d6189c5a2b40b91e3690c1c3350375410e5702
